docs/v3tov4.md

```diff
@@ -0,0 +1,283 @@
+# Upgrading from GoSDK v3.x to GoSDK v4.x
+
+Support for older versions have been removed. Therefore, this new version is not compatible with the previous one. 
+This document describes the changes between versions and how to adapt an integration built with the GoSDK v3.x to the 
+new one.
+
+The Go SDK v4 contains the following changes:
+
+* A new JSON schema version 4.
+* Add support for dimensional metrics using the [Metrics API format][1].
+* New metric data types: `count` and `summary`.
+* Removed support for `RATE`, `PRATE` and `DELTA` metric types.
+* Local entities have been removed.
+* Metrics, Events and Inventory can be attached either to Entities or an Integration.
+* Support for Go Modules
+* Removed support for protocols prior to v4.x.
+
+## JSON schema changes
+
+The following section explains the new JSON schema. Be aware that the new SDK only supports this new JSON document. 
+These are the most important changes:
+
+* A new "integration" object at the top-level.
+
+* The "entity" and "metrics" objects have been modified.
+
+### JSON protocol v4 sample
+```
+{
+  "protocol_version":"4",                      # protocol version number
+  "integration":{                              # this data will be added to all metrics and events as attributes,                                               
+                                               # and also sent as inventory.
+    "name":"integration name",
+    "version":"integration version"
+  },
+  "data":[                                    # List of entities
```

Comment:
The comment says List of entities, should be List of objects containing entities, metrics, events & inventory.